src/posit/connect/resources.py

```diff
@@ -0,0 +1,31 @@
+from abc import ABC, abstractmethod
+from typing import Generic, List, Optional, TypeVar
+
+
+T = TypeVar("T")
+
+
+class Resources(ABC, Generic[T]):
+    @abstractmethod
+    def create(self, *args, **kwargs) -> T:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def delete(self, *args, **kwargs) -> None:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def find(self, *args, **kwargs) -> List[T]:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def find_one(self, *args, **kwargs) -> Optional[T]:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def get(self, *args, **kwargs) -> T:
```

Comment:
Should `id`/`guid` be a part of the `get()` signature?

Reply:
Maybe? My assumption here is that certain may want to use other argument parameters.